src/core/mathsEvaluator.js

```diff
@@ -54,16 +53,60 @@ const defaultDecimalConfig = {
 };
 
 /**
- * List of config entries the Decimal constructor accepts
- * @type {string[]}
+ * Checks if an object is empty.
+ * @param {*} obj - The object to check.
+ * @returns {boolean} - True if the object is empty, false otherwise.
  */
-const decimalConfigEntries = ['precision', 'rounding', 'toExpNeg', 'toExpPos', 'maxE', 'minE', 'modulo', 'crypto'];
+const isEmpty = obj => [Object, Array].includes((obj || {}).constructor) && !Object.entries(obj || {}).length;
 
 /**
- * List of config entries the Parser constructor accepts
- * @type {string[]}
+ * Checks if an object is a plain object.
+ * @param {*} obj - The object to check.
+ * @returns {boolean} - True if the object is a plain object, false otherwise.
  */
-const parserConfigEntries = ['operators'];
+const isPlainObject = obj => [Object].includes((obj || {}).constructor) && Object.entries(obj || {}).length;
+
+/**
+ * Create a new function that calls func with args set on first place.
+ * @param {function} func - The function to partially apply.
+ * @param  {...*} boundArgs - The arguments to partially apply before.
+ * @returns {function} - The partially applied function.
+ */
+function partial(func, ...boundArgs) {
```

Comment:
question: looks like something like usually acheved with `.bind`. This is an implementation to avoid any `this` hassle?

Reply:
Yes, it could be done with `.bind` as well. However, it would be more verbose, and actually, the idea was to have a straightforward replacement for lodash.